Bio/Seq.py

```diff
@@ -1392,41 +1392,73 @@ def count_overlap(self, sub, start=0, end=sys.maxsize):
         return end - start - len_sub_str + 1
 
     def complement(self):
-        """Return the complement of an unknown nucleotide equals itself.
+        """Return the complement assuming it is DNA.
 
-        >>> my_nuc = UnknownSeq(8)
+        In typical usage this will return the same unknown sequence:
+
+        >>> my_nuc = UnknownSeq(8, character='N')
         >>> my_nuc
-        UnknownSeq(8, character='?')
+        UnknownSeq(8, character='N')
         >>> print(my_nuc)
-        ????????
+        NNNNNNNN
         >>> my_nuc.complement()
-        UnknownSeq(8, character='?')
+        UnknownSeq(8, character='N')
         >>> print(my_nuc.complement())
-        ????????
+        NNNNNNNN
+
+        It would be unexpected to use a nucleotide letter other than N, but if
+        you do the complement base is used:
+
+        >>> UnknownSeq(8, character="A").complement()
+        UnknownSeq(8, character='T')
         """
-        return self
+        s = Seq(self._character).complement()
+        return UnknownSeq(self._length, character=str(s))
 
     def complement_rna(self):
-        """Return the complement assuming it is RNA."""
-        return self.complement()
+        """Return the complement assuming it is RNA.
+
+        In typical usage this will return the same unknown sequence. It would
+        be unexpected to use a nucleotide letter other than N, but if you do,
+        the complement base is used:
```

Comment:
Same here. The name `UnknownSeq` is a bit unfortunate.

Reply:
Unexpected because it wouldn't be unknown. In hindsight perhaps something about homopolymer might have been a better name...